gammapy/scripts/cta_sensitivity.py

```diff
@@ -0,0 +1,289 @@
+
+        Parameters
+        ----------
+        excess_counts : `~numpy.ndarray`
+            Array of gamma-ray excess (bins in reconstructed energy)
+        model : `~gammapy.spectrum.models.SpectralModel`
+            Spectral model
+        aeff : `~gammapy.irf.EffectiveAreaTable`
+            Effective area
+        edisp : `~gammapy.irf.EnergyDispersion2D`
+            Energy dispersion
+
+        Returns
+        -------
+        flux : `~astropy.units.Quantity`
+            Array of 1TeV fluxes (bins in reconstructed energy)
+        """
+        # Compute expected excess
+        predictor = CountsPredictor(model, aeff=aeff, edisp=edisp, livetime=self.livetime)
+        predictor.run()
+        counts = predictor.npred.data.data.value
+
+        # Conversion in flux
+        flux = excess_counts / counts * u.Unit('cm-2 s-1 TeV-1')
+
+        return flux
+
+    def run(self):
+        """Do the computation."""
+        # Creation of the spectral shape
+        norm = 1 * u.Unit('cm-2 s-1 TeV-1')
+        index = self.slope
+        ref = 1 * u.TeV
+        model = PowerLaw(index=index, amplitude=norm, reference=ref)
+
+        # Get the bins in reconstructed  energy
+        reco_energy = self.irf.bkg.energy
+
+        # Start the computation
+        bkg_counts = self.get_bkg(self.irf.bkg)
+        excess_counts = self.get_excess(bkg_counts)
+
+        phi_0 = self.get_1TeV_differential_flux(excess_counts, model, self.irf.aeff, self.irf.rmf)
+        energy = reco_energy.log_center()
+        dnde_model = model.evaluate(energy=energy, index=index, amplitude=1, reference=ref)
+        diff_flux = (phi_0 * dnde_model * energy ** 2 * u.TeV.to('erg')).to('erg / (cm2 s)')
+
+        self.energy = reco_energy.log_center()
+        self.diff_sens = diff_flux
+
+    @property
+    def diff_sensi_table(self):
+        """Differential sensitivity table (`~astropy.table.Table`)."""
+        table = Table()
+        table['ENERGY'] = Column(self.energy, unit=self.energy.unit,
+                                 description='Energy')
+        table['FLUX'] = Column(self.diff_sens, unit=self.diff_sens.unit,
+                               description='Differential flux')
+        return table
+
+    @property
+    def _ref_diff_sensi(self):
+        """Reference differential sensitivity table (`~astropy.table.Table`)"""
+        table = Table()
+        energy = self.irf.sens.energy.log_center()
+        table['ENERGY'] = Column(energy, unit=energy.unit,
+                                 description='Energy')
+        flux = self.diff_sens
+        table['FLUX'] = Column(flux, unit=flux.unit,
+                               description='Differential flux')
+        return table
+
+    def print_results(self):
+        """Print results to the console."""
+        self.diff_sensi_table.pprint()
+
+        if log.getEffectiveLevel() == 10:
+            self._ref_diff_sensi.pprint()
+
+    def plot(self, ax=None):
```

Comment:
Don't know how it is handled elsewhere in gammapy but maybe it makes sense to return the axis object here?
Also might it be better to let the user call `plt.show()` ? I might want to save it using `savefig` or something.